### PR TITLE
Set a limit of 10000 for index

### LINKED ISSUE #226: Index is above MAX_INT will return invalid command
Proposed solution: change the index to say that it is within 10,000

### PR DESCRIPTION
**Pull request by Droffilc13**
fixes #226 

Set a hard limit of 10,000 for index. This is reflected in the invalid command format message

### CODE REVIEW

File: src/test/java/seedu/address/commons/util/StringUtilTest.java [hunk 1]
```diff
@@ -45,6 +45,86 @@ public void isNonZeroUnsignedInteger() {
...
+        assertTrue(StringUtil.isIntegerUpperBounded(Integer.toString(validLimit - 1), validLimit));
+
+        // EP: number equal to limit
+        assertTrue(StringUtil.isIntegerUpperBounded(Integer.toString(validLimit), validLimit));
+
+        // EP: number above limit
+        assertFalse(StringUtil.isIntegerUpperBounded(Integer.toString(validLimit + 1), validLimit));
+
+        // EP: numbers with white space
+        assertFalse(StringUtil.isIntegerUpperBounded(" 10 ", validLimit)); // Leading/trailing spaces
```
Comment (qinguorui2001) [inline src/test/java/seedu/address/commons/util/StringUtilTest.java:81]: which kind of exception will occur if there are leading or trailing spaces for index

File: src/test/java/seedu/address/commons/util/StringUtilTest.java [hunk 2]
```diff
@@ -45,6 +45,86 @@ public void isNonZeroUnsignedInteger() {
...
+        assertFalse(StringUtil.isIntegerUpperBounded("1 0", validLimit)); // Spaces in the middle
+
+        // EP: number larger than Integer.MAX_VALUE or number smaller than Integer.MIN_VALUE -> overflow
+        assertTrue(StringUtil.isIntegerUpperBounded(Long.toString(Integer.MAX_VALUE + 1), validLimit));
```
Comment (qinguorui2001) [inline src/test/java/seedu/address/commons/util/StringUtilTest.java:85]: boundary value? if so label it

File: src/test/java/seedu/address/commons/util/StringUtilTest.java [hunk 3]
```diff
@@ -45,6 +45,86 @@ public void isNonZeroUnsignedInteger() {
...
+        // EP: empty strings
+        assertFalse(StringUtil.isNonZeroUnsignedIntegerBelowLimit("", validLimit)); // Boundary value
+        assertFalse(StringUtil.isNonZeroUnsignedIntegerBelowLimit("  ", validLimit));
+
+        // EP: not a number
+        assertFalse(StringUtil.isNonZeroUnsignedIntegerBelowLimit("a", validLimit));
+        assertFalse(StringUtil.isNonZeroUnsignedIntegerBelowLimit("aaa", validLimit));
+
+        // EP: zero
+        assertFalse(StringUtil.isNonZeroUnsignedIntegerBelowLimit("0", validLimit));
```
Comment (qinguorui2001) [inline src/test/java/seedu/address/commons/util/StringUtilTest.java:104]: also boundary?

**Review by qinguorui2001 — CHANGES_REQUESTED** [review]

File: src/test/java/seedu/address/commons/util/StringUtilTest.java (hunk 1 above)
Comment (Droffilc13) [reply src/test/java/seedu/address/commons/util/StringUtilTest.java:81]: ```NumberParseException```, same with all other invalid strings.
The method will just catch this exception and return false because the string is an invalid format

File: src/test/java/seedu/address/commons/util/StringUtilTest.java (hunk 3 above)
Comment (Droffilc13) [reply src/test/java/seedu/address/commons/util/StringUtilTest.java:104]: Good point. I have taken note of your boundary value comments. Made several edits to a few more methods with boundary values as well.

**Review by qinguorui2001 — APPROVED** [review]
LGTM
